components/core/src/os/filesystem/linux.rs

```diff
@@ -15,11 +15,49 @@
 use libc::{self, c_int, c_char, mode_t};
 
 pub use std::os::unix::fs::symlink;
+use std::ffi::CString;
 
-pub fn chown(r_path: *const c_char, uid: u32, gid: u32) -> c_int {
-    unsafe { libc::chown(r_path, uid, gid) }
+use error::{Result, Error};
+
+fn validate_raw_path(path: &str) -> Result<*mut c_char> {
+    let c_path = match CString::new(path) {
+        Ok(c) => c,
+        Err(e) => {
+            return Err(Error::PermissionFailed(format!("Can't create string from path {:?}: {}",
+                                                       path,
+                                                       e)))
+        }
+    };
+    Ok(c_path.into_raw())
```

Comment:
yeah its returned to both the linux chown and chmod functions that need the raw pointer. This function allows this to be done in one place.